### DISCUSSION TITLE
Override implementation of a script in a specific package

### DISCUSSION THREAD

**Discussion by derrik-fleming** [invertase/melos#805] [Q&A]
Say I have three packages, `A`, `B`, and `C`. I want a script `build` defined at the top-level. For packages `A` and `B` the implementation of this script is identical, but if I were to run the script on `C` it would fail. I still need a `build` script for `C` but it isn't the same implementation as `C`.

Coming from other similar tools I've used, they offer the ability to override the implementation of `build` in package `C` by placing their equivalent of a `melos.yaml` at the root of package `C`.

When executing `melos build` from the repo root `build`, as it's defined in `./melos.yaml`, would run for packages `A` and `B` but for package `C` it would source  `build`'s definition from `./packages/C/melos.yaml`.

I'm aware of `packageFilters`, for my specific scenario they seem less than ideal. Is there a better way supported by Melos?

**Comment by spydon** [comment]
I don't think we'd want to extend the scripting capabilities by introducing new files.
For these cases I would just have a script with two steps with different package filters.

If a good override syntax contained within the root `melos.yaml` file that could be considered I think.
